src/v/compression/tests/zstd_tests.cc

```diff
@@ -85,6 +86,47 @@ SEASTAR_THREAD_TEST_CASE(stream_zstd_test) {
     }
 }
 
+SEASTAR_THREAD_TEST_CASE(async_stream_zstd_test) {
+    compression::async_stream_zstd fn;
+    auto test_sizes = get_test_sizes();
+    for (size_t i : sizes) {
+        iobuf buf = gen(i);
+
+        auto cbuf = fn.compress(buf.share(0, i)).get();
+        auto dbuf = fn.uncompress(std::move(cbuf)).get();
+
+        BOOST_CHECK_EQUAL(dbuf, buf);
+    }
+}
```

Comment:
how about also expanding `zstd_stream_bench`? it would be nice too if it was setup to trigger a reactor stall without the async version (now that seastar has a 25 ms threshold it should be easier to trigger). seeing the difference in performance will be interesting.

Reply:
Will do